wine-utils@elblake/files/wine-utils@elblake/applet.js

```diff
@@ -0,0 +1,450 @@
+                }
+            ));
+        } catch (e) {
+            global.log('ERROR: ' + e.toString());
+        }
+    },
+
+    remove_trailing_slash: function (dir) {
+        while (dir.substr(-1) === '/') {
+            dir = dir.substr(0, dir.length-1);
+        }
+        return dir;
+    },
+
+    expand_path: function (a) {
+        let char1 = a.substr(0,1);
+        if ((char1 === '~') && (a.substr(1,1) === '/')) {
+            // A file path where the squiggle will be substituted with
+            // the home directory.
+            let dir = this.remove_trailing_slash(GLib.get_home_dir());
+            a = dir + a.substr(1);
+            char1 = a.substr(0,1);
+        }
+        if (char1 !== '/') {
+            // A file path probably relative from the home directory.
+            let dir = this.remove_trailing_slash(GLib.get_home_dir());
+            a = dir + '/' + a;
+        }
+        return this.remove_trailing_slash(a);
+    },
+
+    parse_list: function (text) {
+        let list = [];
+        let lines = text.split('\n');
+        for (let i = 0; i < lines.length; ++i) {
+            let a = lines[i].trim();
+            if (a === '') {
+                continue;
+            }
+            if (a.substr(0,1) !== '#') {
+                let ret = a.split(':', 2);
+                let ret_1 = ret[1].trim();
+                if (ret.length == 2) {
+                    list.push([ret[0].trim(), ret_1]);
+                } else if (ret.length == 1) {
+                    list.push(['', ret_1]);
+                }
+            }
+        }
+        return list;
+    },
+
+    parse_icon: function (name) {
+        let arr = name.match(/^(.+)\s+\(icon:([^()]+)\)\s*$/);
+        if ((arr !== null) && (arr.length > 2)) {
+            return [arr[1].trim(), arr[2].trim()];
+        } else {
+            return [name, ''];
+        }
+    },
+
+    rebuild_menu: function () {
+        this.menu.removeAll();
+
+        if (this.prefixFileList !== '') {
+            this.menu_prefixes = this.menu_tools(_('Prefixes'), []);
+            this.menu.addMenuItem(new PopupMenu.PopupSeparatorMenuItem());
+        } else {
+            this.menu_prefixes = null;
+        }
+        
+        if (this.showRunCommand) {
+            this.menu_new_item(
+                this,
+                _("Run Command..."),
+                'none',
+                Lang.bind(this, this.run_command));
+        }
+        
+        if (this.showCommandPrompt) {
+            this.menu_new_item(
+                this,
+                _("Command Prompt"),
+                'wine-cmd',
+                Lang.bind(this, this.use_cmd));
+        }
+
+        if (this.showAdminProgramsMenu) {
+            this.menu_tools(_('Configuration'), this.menu_to_binds(
+                this.admin_programs_list));
+        }
+
+        if (this.showProgramsMenu) {
+            this.menu_programs = this.menu_tools(_('Programs'), []);
```

Comment:
Just a note that these functions will not function without the proper translation support (importing Gettext). For the applet code, you can look at other applets for an example. For the bash scripts, there are a decent volume of examples if you look at the [cinnamon-spices-actions](https://github.com/linuxmint/cinnamon-spices-actions) repository. After both of these are done, there is a script in the root of the repository that can be used to generate the translation template file that translators can use: `./cinnamon-spices-makepot wine-utils@elblake`

This looks like it will at least work for English-speaking users as-is. Would you rather see this merged now and implement the proper translation support after the fact or should this wait for the translation support before you want to have it merged?

Reply:
Ah I didn't know about the Gettext part, I'll add that in right away. I'll make the needed changes for the shell scripts as well too.

Reply:
I think I've made the needed changes.

Reply:
Well done!